src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -4,42 +4,162 @@
+#define ENSGORO_BOMBBUYFLAG_TALKED_ZORA_FINALNIGHT 0x0080
+#define ENSGORO_BOMBBUYFLAG_TALKED_GORON 0x0100
+#define ENSGORO_BOMBBUYFLAG_TALKED_GORON_FINALNIGHT 0x0200
+#define ENSGORO_BOMBBUYFLAG_YESBUY 0x0400
+
+#define SCENEFLAG_TORCHES_LIT Flags_GetSwitch(playState, ((this->actor.params & 0x7F0) >> 4))
+#define CHECK_WEEKEVENT(idx, bitmask) (gSaveContext.save.weekEventReg[idx] & bitmask)
+#define SET_WEEKEVENT(idx, bitmask) gSaveContext.save.weekEventReg[idx] |= bitmask
+
+#define CHECK_WEEKEVENT_GK_PACIFIED \
+    CHECK_WEEKEVENT(0x16, 0x04) // Kid is asleep. Defo animation of shrine gorons set to shivering.
+#define CHECK_WEEKEVENT_SPRING_TALK_JG \
+    CHECK_WEEKEVENT(0x4D, 0x80) // You have talked to the Elder as goron while it is spring.
+
+#define CHECK_WEEKEVENT_SGOROA_GKLOUD_NONGORON CHECK_WEEKEVENT(0x24, 0x01) // Talked to shrine A (non Goron), GK bawling
+#define CHECK_WEEKEVENT_SGOROA_GKLOUD_GORON CHECK_WEEKEVENT(0x24, 0x02)    // Talked to shrine A (as Goron), GK bawling
+#define CHECK_WEEKEVENT_SGOROA_GKQUIET_NONGORON \
+    CHECK_WEEKEVENT(0x24, 0x04) // Talked to shrine A (non Goron), GK quiet, torches unlit
+#define CHECK_WEEKEVENT_SGOROA_GKQUIET_GORON \
+    CHECK_WEEKEVENT(0x24, 0x08) // Talked to shrine A (as Goron), GK quiet, torches unlit
+#define CHECK_WEEKEVENT_SGOROA_GKQUIET_TORCHES \
+    CHECK_WEEKEVENT(0x24, 0x10) // Talked to shrine A (any form), GK quiet, torches lit
+#define CHECK_WEEKEVENT_SGOROB_GKLOUD_NONGORON CHECK_WEEKEVENT(0x24, 0x20) // Talked to shrine B (non Goron), GK bawling
+#define CHECK_WEEKEVENT_SGOROB_GKLOUD_GORON CHECK_WEEKEVENT(0x24, 0x40)    // Talked to shrine B (as Goron), GK bawling
+#define CHECK_WEEKEVENT_SGOROB_GKQUIET_NONGORON CHECK_WEEKEVENT(0x24, 0x80) // Talked to shrine B (non Goron), GK quiet
+#define CHECK_WEEKEVENT_SGOROB_GKQUIET_GORON CHECK_WEEKEVENT(0x25, 0x01)    // Talked to shrine B (as Goron), GK quiet
+#define CHECK_WEEKEVENT_SGOROC_GKLOUD_GORON CHECK_WEEKEVENT(0x25, 0x02)     // Talked to shrine C (as Goron), GK bawling
+#define CHECK_WEEKEVENT_SGOROC_GKQUIET_GORON CHECK_WEEKEVENT(0x25, 0x04)    // Talked to shrine C (as Goron), GK quiet
+#define SET_WEEKEVENT_SGOROA_GKLOUD_NONGORON SET_WEEKEVENT(0x24, 0x01)
+#define SET_WEEKEVENT_SGOROA_GKLOUD_GORON SET_WEEKEVENT(0x24, 0x02)
+#define SET_WEEKEVENT_SGOROA_GKQUIET_NONGORON SET_WEEKEVENT(0x24, 0x04)
+#define SET_WEEKEVENT_SGOROA_GKQUIET_GORON SET_WEEKEVENT(0x24, 0x08)
+#define SET_WEEKEVENT_SGOROA_GKQUIET_TORCHES SET_WEEKEVENT(0x24, 0x10)
+#define SET_WEEKEVENT_SGOROB_GKLOUD_NONGORON SET_WEEKEVENT(0x24, 0x20)
+#define SET_WEEKEVENT_SGOROB_GKLOUD_GORON SET_WEEKEVENT(0x24, 0x40)
+#define SET_WEEKEVENT_SGOROB_GKQUIET_NONGORON SET_WEEKEVENT(0x24, 0x80)
+#define SET_WEEKEVENT_SGOROB_GKQUIET_GORON SET_WEEKEVENT(0x25, 0x01)
+#define SET_WEEKEVENT_SGOROC_GKLOUD_GORON SET_WEEKEVENT(0x25, 0x02)
+#define SET_WEEKEVENT_SGOROC_GKQUIET_GORON SET_WEEKEVENT(0x25, 0x04)
+
+#define ACTOR_USAGE_CONTEXT (this->actor.params & 0x000F)
```

Comment:
For params usage like this, we make a macro in the `.h` file that takes in the base actor pointer `thisx`.
For example this would be 
`ENSGORO_USAGE_CONTEXT(thisx) ((thisx)->params & 0xF)`
Note this applies to all usage of `params` that splits it up in someway.

Reply:
Also I'm not quite sure what you mean by `USAGE_CONTEXT` is it like the TYPE of goron?